fix(DatabaseModal): Resolve Connect button issue for SQLAlchemy URI database connections

EnxDev opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
This PR addresses a bug where the Connect button remains disabled when attempting to connect a database using the SQLAlchemy URI flow in the DatabaseModal.

**Problem**  Fixes #34110
When selecting "Connect this database with a SQLAlchemy URI string instead", entering a valid URI does not enable the Connect button.

This was likely introduced in #33826, which added stricter validation that does not currently account for the URI-based flow.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
- Before 

https://github.com/user-attachments/assets/ced5e02a-446f-4a02-ab1c-6008f6fbff74


- After

https://github.com/user-attachments/assets/e0a7205e-c2d5-498a-9a42-9f0f907e98f4

- Before 
<img width="778" height="1024" alt="before-1" src="https://github.com/user-attachments/assets/0aee30ee-58a4-441e-8718-3a08a6d1b2cf" />

- After
<img width="814" height="1012" alt="after-1" src="https://github.com/user-attachments/assets/0cf8193d-6ac1-4bc1-9703-5b3f68d857ae" />

- Before
<img width="820" height="1031" alt="before-2" src="https://github.com/user-attachments/assets/078b4f77-3e5c-4e4b-9b64-2497fc1919ac" />

- After
<img width="756" height="1012" alt="after-2" src="https://github.com/user-attachments/assets/5cc3b7f3-8494-4f56-b98f-3e8d52b49192" />
<img width="583" height="164" alt="after-2 3" src="https://github.com/user-attachments/assets/73f24419-6ba4-4647-a00a-5b3aa3d6fd89" />


### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->

**test  1**
- Choose PostgreSQL
- Do not fill all fields
- Click on "Connect this database with a SQLAlchemy URI string instead"
- Insert a valid SQLAlchemy URI
- Click "Test connection"
- The "Connect" button should be enabled
- Modify the "SQLAlchemy URI" field with an invalid URI
- The "Connect" button should be disabled
- Click "Test connection"
- The "Connect" button should remain disabled

**test 2**
- Click "Back"
- Choose PostgreSQL
- The "Connect" button should be disabled
- Fill in all required fields
- The "Connect" button should be enabled
- Click "Connect"
- The connected database should appear in the list

**test 3**
- Click "+ Database"
- Choose PostgreSQL
- Fill in all fields with previously used values
- The "Connect" button should be enabled
- Click "Connect"
- A "Database already exists" error should appear
- Change the "Display Name" field
- Click "Connect"
- The connected database should appear in the list

**test 4**
- Click "+ Database"
- Choose PostgreSQL
- Click on "Connect this database with a SQLAlchemy URI string instead"
- Fill in the "Display Name"
- Insert a valid "SQLAlchemy URI"
- Click "Connect"
- The connected database should appear in the list

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [x] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

EnxDev committed:
fix(DatabaseModal): enable connect button after successful test using SQLAlchemy URI

serverdevil commented:
@EnxDev While at it, I'd like to report another thing that I noticed while playing around with this. The validation status is retained if the user goes back (Back button) and changes the source db. 
E.g. I select DB type A, enter the details and "Connect" button enables. Instead of clicking "Connect" I choose "Back", select DB type B, the modal already shows the "Connect" button enabled.

EnxDev commented:
> @EnxDev While at it, I'd like to report another thing that I noticed while playing around with this. The validation status is retained if the user goes back (Back button) and changes the source db. E.g. I select DB type A, enter the details and "Connect" button enables. Instead of clicking "Connect" I choose "Back", select DB type B, the modal already shows the "Connect" button enabled.

Yes, I’ve noticed that as well and I’m currently running various tests to address it.
Thank you very much for pointing it out; I'll be working on fixing everything shortly

Please let me know if you find anything else, thanks!

EnxDev committed:
refactor(DatabaseModal): improve error state handling and validation in DatabaseModal
- Improved error state management and field validation
- Updated modal close icon for better UX
- Fixed spacing in DatabaseModal action buttons
- Adjusted margins in error message description

EnxDev committed:
refactor(DatabaseModal): handle additional validation case for SQLAlchemy URI input

EnxDev committed:
fix RTL test

EnxDev committed:
fix PropertiesModal test
